.github/workflows/build-system.yml

```diff
@@ -0,0 +1,99 @@
+name: Continuous Integration v2
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  CI:
+    name: ${{ matrix.name }} - ${{ matrix.configuration }}
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [RelWithDebInfo, MinSizeRel]
+        os: [windows-latest, windows-2019, ubuntu-latest, macos-10.15]
+        include:
+        - os: ubuntu-latest
+          container: novelrt/novelrt-build:latest
+          profile: linux-clang10-amd64
+          name: Ubuntu - Clang 10
+        - os: windows-2019
+          profile: windows-vs2019-amd64
+          name: Windows - Visual Studio 2019
+        - os: windows-latest
+          profile: windows-vs2022-amd64
+          name: Windows - Visual Studio 2022
+        - os: macos-10.15
+          profile: macos-appleclang12-amd64
+          name: macOS 10.15 - Apple-Clang 12
+          configuration: Debug
+        - os: macos-10.15
+          profile: macos-appleclang12-amd64
+          name: macOS 10.15 - Apple-Clang 12
+          configuration: Release
+        exclude:
+        - os: macos-10.15
+          configuration: RelWithDebInfo
+        - os: macos-10.15
+          configuration: MinSizeRel
```

Comment:
macOS builds were taking over an hour, so I exclude them and re-include them under normal `Release` and `Debug` configs.

Takes ~5 minutes 😄

Reply:
That's a :harold: moment if I've ever heard one. If I do the dockerfile crap this should simplify things a LOT.